scripts/tests/test_validate_unwanted_patterns.py

```diff
@@ -0,0 +1,423 @@
+# with pytest.raises(ValueError):
+#    pass
+"""
+            ),
+            [],
+        ),
+        (
+            (
+                """
+with pytest.raises(
+    ValueError,
+    match="foo"
+):
+    pass
+"""
+            ),
+            [],
+        ),
+        (
+            (
+                """
+with pytest.raises(ValueError):
+    pass
+"""
+            ),
+            [
+                (
+                    1,
+                    (
+                        "Bare pytests raise have been found. "
+                        "Please pass in the argument 'match' "
+                        "as well the exception."
+                    ),
+                ),
+            ],
+        ),
+        (
+            (
+                """
+with pytest.raises(ValueError, match="foo"):
+    with pytest.raises(ValueError):
+        pass
+    pass
+"""
+            ),
+            [
+                (
+                    2,
+                    (
+                        "Bare pytests raise have been found. "
+                        "Please pass in the argument 'match' "
+                        "as well the exception."
+                    ),
+                ),
+            ],
+        ),
+        (
+            (
+                """
+with pytest.raises(ValueError):
+    with pytest.raises(ValueError, match="foo"):
+        pass
+    pass
+"""
+            ),
+            [
+                (
+                    1,
+                    (
+                        "Bare pytests raise have been found. "
+                        "Please pass in the argument 'match' "
+                        "as well the exception."
+                    ),
+                ),
+            ],
+        ),
+        (
+            (
+                """
+with pytest.raises(
+    ValueError
+):
+    pass
+"""
+            ),
+            [
+                (
+                    1,
+                    (
+                        "Bare pytests raise have been found. "
+                        "Please pass in the argument 'match' "
+                        "as well the exception."
+                    ),
+                ),
+            ],
+        ),
+        (
+            (
+                """
+with pytest.raises(
+    ValueError,
+    # match = "foo"
+):
+    pass
+"""
+            ),
+            [
+                (
+                    1,
+                    (
+                        "Bare pytests raise have been found. "
+                        "Please pass in the argument 'match' "
+                        "as well the exception."
+                    ),
+                ),
+            ],
+        ),
+    ],
+)
+def test_pytest_raises(data, expected):
+    fd = io.StringIO(data.strip())
+    result = list(validate_unwanted_patterns.bare_pytest_raises(fd))
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "data, expected",
+    [
+        (
+            'msg = ("bar " "baz")',
+            [
+                (
+                    1,
```

Comment:
IIUC `expected` is always the same, so move to body of test instead of parametrizing

Reply:
Actually ```expected``` is not always the same, One is checking that the check is detecting lines, and the other check that it also check the number of occurrences in each line.

---

For example if you have this:

```python
msg = "foo " "bar"
```

the check will output:
```
path/to/file:1 String unnecessarily split in two....
```

and for this:

```python
msg = "foo " "bar " "baz"
```

it will output:
```
path/to/file:1 String unnecessarily split in two....
path/to/file:1 String unnecessarily split in two....
```